AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -176,13 +178,11 @@ - (void)_layoutNodes:(NSArray *)nodes ofKind:(NSString *)kind atIndexPaths:(NSAr
       dispatch_queue_t queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
       dispatch_apply(batchCount, queue, ^(size_t i) {
         unsigned long k = j + i;
-        ASCellNodeBlock cellBlock = nodes[k];
-        ASCellNode *node = cellBlock();
+        ASIndexedNodeContext *context = contexts[k];
+        ASCellNodeBlock nodeBlock = context.nodeBlock;
+        ASCellNode *node = nodeBlock();
```

Comment:
Good point. We should also nil out the context's nodeBlock immediately after executing it.